mewarpx/tests/test_custom_criteria.py

```diff
@@ -0,0 +1,108 @@
+    ("name"),
+    [
+        'Run2D',
+        'Run2D_RZ',
+        'Run3D'
+    ]
+)
+def test_capacitive_discharge_multigrid(capsys, name):
+    basename = "Run"
+    use_rz = 'RZ' in name
+    dim = int(name.replace(basename, '')[0])
+
+    # Initialize and import only when we know dimension
+    mwxutil.init_libwarpx(ndim=dim, rz=use_rz)
+    from mewarpx import testing_util
+    from mewarpx.setups_store import diode_setup
+    from mewarpx import mwxrun
+    from pywarpx import callbacks
+
+    # Include a random run number to allow parallel runs to not collide. Using
+    # python randint prevents collisions due to numpy rseed below
+    #testing_util.initialize_testingdir(name)
+
+    # Initialize each run with consistent, randomly-chosen, rseed. Use a random
+    # seed instead for initial dataframe generation.
+    # np.random.seed()
+    np.random.seed(92160881)
+
+    sim = mwxrun.mwxrun.simulation
+
+    # Specific numbers match older run for consistency
+    FREQ = 13.56e6  # MHz
+    DT = 1.0 / (400 * FREQ)
+    DIAG_STEPS = 2
+    DIAG_INTERVAL = DIAG_STEPS*DT
+    VOLTAGE = 450.0
+    D_CA = 0.067  # m
+    run = diode_setup.DiodeRun_V1(
+        dim=dim,
+        rz=use_rz,
+        V_ANODE_CATHODE=VOLTAGE,
+        V_ANODE_EXPRESSION="%.1f*sin(2*pi*%.5e*t)" % (VOLTAGE, FREQ),
+        D_CA=D_CA,
+        INERT_GAS_TYPE='He',
+        N_INERT=9.64e20,  # m^-3
+        T_INERT=300.0,  # K
+        PLASMA_DENSITY=2.56e14,  # m^-3
+        T_ELEC=30000.0,  # K
+        NX=16,
+        NZ=128,
+        # This gives equal spacing in x & z
+        PERIOD=D_CA * 16 / 128.0,
+        DT=DT,
+        TOTAL_TIMESTEPS=10,
+        DIAG_STEPS=DIAG_STEPS,
+        DIAG_INTERVAL=DIAG_INTERVAL,
+        NUMBER_PARTICLES_PER_CELL=[16, 32],
+        FIELD_DIAG_DATA_LIST=['rho_electrons', 'rho_he_ions', 'phi'],
+        SIMULATION=sim
+    )
+    # Only the functions we change from defaults are listed here
+    run.setup_run(
+        init_conductors=False,
+        init_scraper=False,
+        init_injectors=False,
+        init_inert_gas_ionization=True,
+        init_field_diag=True,
+        init_simcontrol=True,
+        init_warpx=True
+    )
+    run.control.add_checker(check_particle_nums)
+
+    callbacks.installafterstep(add_particles)
+
+    # Run the main WARP loop
+    while run.control.check_criteria():
+        run.SIMULATION.step(1)
+
+    #######################################################################
+    # Cleanup and final output                                            #
+    #######################################################################
+
+    # TODO: what can be kept here between these two lines?
+    # these are commented out because I don't know what to pass in for capsys
+    # run.runresults.finalize_save()
+    # out, _ = capsys.readouterr()
+
+    # print(out)
+    # # make sure out isn't empty
+    # assert out
+# this is how I'm running it
+test_capacitive_discharge_multigrid(None, "Run2D")
```

Comment:
@MKieburtz I'm guessing this is your problem right now. `pytest` automatically runs functions that begin with `test`, and in general modules should *not* execute code like this when they're just imported. Sorry, I fixed it last minute in the main diode template and forgot to point it out to you, but see everything below line 84 https://github.com/ModernElectron/WarpX/blob/b2a85ee6fb43e4a58a4f616388f5e74f33c118ae/mewarpx/tests/test_capacitive_discharge.py#L84 - the `if __name__` block is what solves your problem here; it runs the test if you call it with `python` but not if you call it with `pytest`.